### DISCUSSION TITLE
Maintain multiple local repositories with different content pushed to the same GitHub repo?

### DISCUSSION THREAD

**Discussion by p-et** [community/community#45315] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

This is my data/ local repository situation:
- I have a project consisting of more than one (1) programming language. 
- **During development** the code for each programming language is **stored to different local areas** (i.e. different local servers). 
- **After development** (i.e. once a program is considered final through formal validation) all code of all used program languages will be **saved to one local area**.

**During development** I have two (2) local sources I want to watch with **git** and push to GitHub, so the **combination of both local repositories** defines the complete pushed repository. 

Note: Once a program is final there is no need to push from the final local repository. In case changes are needed the development cycle will re-start and I will have 2 local repositories again.

My questions:
1. What is the best way/ best pragmatic approach to deal with this situation?
2. Can I combine two (2) local repositories using **push** into one GitHub repository? (No merging).
3. If question 2 = _yes_ - how can I achieve this using **git** commands on the teminal?

**Comment by airtower-luna** [comment, marked as answer]
> Can I combine two (2) local repositories using push into one GitHub repository? (No merging).

Without merging (or rebasing to similar effect) that's a no. If you want them in one repository and on the same branch you need a merged history.

Merging would be exactly the Git way of dealing with this situation. Create as many clones of the repository you want (for example one per language), work on them on a branch each, once one of those is ready merge it. If you only want to avoid merge commits you can rebase instead of merging, but you need a combined history.

You can use [sparse checkout](https://git-scm.com/docs/git-sparse-checkout) to put only certain files/directories into the working tree of a local repository, but:

1. That's marked as experimental, and
2. the other parts of the repository are still part of the history.
